espnet/nets/pytorch_backend/transformer/decoder_layer.py

```diff
@@ -33,10 +45,33 @@ def forward(self, tgt, tgt_mask, memory, memory_mask):
         :param torch.Tensor memory: encoded source features (batch, max_time_in, size)
         :param torch.Tensor memory_mask: mask for memory (batch, max_time_in)
         """
-        x = tgt
-        nx = self.norm1(x)
-        x = x + self.dropout(self.self_attn(nx, nx, nx, tgt_mask))
-        nx = self.norm2(x)
-        x = x + self.dropout(self.src_attn(nx, memory, memory, memory_mask))
-        nx = self.norm3(x)
-        return x + self.dropout(self.feed_forward(nx)), tgt_mask, memory, memory_mask
+        residual = tgt
```

Comment:
I just want to make sure.
Is that OK?
`residual ` may not be changed?

Reply:
I checked.

```
[ins] In [1]: import torch

[ins] In [2]: x = torch.zeros(3)

[ins] In [3]: x
Out[3]: tensor([0., 0., 0.])

[ins] In [4]: y = x

[ins] In [5]: x = x + 1

[ins] In [6]: y
Out[6]: tensor([0., 0., 0.])

[ins] In [7]: x
Out[7]: tensor([1., 1., 1.])
```

Reply:
OK. Thanks.